torch/_dynamo/variables/builtin.py

```diff
@@ -1805,6 +1805,9 @@ def check_type(ty):
                 isinstance_type.__class__.__instancecheck__(isinstance_type, arg.value)
             )
 
+        if isinstance(arg, variables.UserDefinedExceptionClassVariable):
```

Comment:
Why wouldn't this fall into the `val = issubclass(arg_type, isinstance_type_tuple)` case below, and why would we need to remove `@property` from `UserDefinedExceptionClassVariable.python_type`?

Reply:
> Why wouldn't this fall into the val = issubclass(arg_type, isinstance_type_tuple) case below,

```python
(Pdb+) arg_type
<class '__main__.CustomException'>
(Pdb+) isinstance_type
<class 'type'>
(Pdb+) isinstance(arg_type, isinstance_type)
True
(Pdb+) issubclass(arg_type, isinstance_type)
False
```

> why would we need to remove @property from UserDefinedExceptionClassVariable.python_type?

`python_type` is not a property.

Reply:
I'm wondering if the isinstance call can have side effects.